src/hotspot/share/classfile/classLoaderDataGraph.cpp

```diff
@@ -501,37 +494,34 @@ bool ClassLoaderDataGraph::is_valid(ClassLoaderData* loader_data) {
 bool ClassLoaderDataGraph::do_unloading() {
   assert_locked_or_safepoint(ClassLoaderDataGraph_lock);
 
-  ClassLoaderData* data = _head;
   ClassLoaderData* prev = nullptr;
   bool seen_dead_loader = false;
   uint loaders_processed = 0;
   uint loaders_removed = 0;
 
-  data = _head;
-  while (data != nullptr) {
+  for (ClassLoaderData* data = _head; data != nullptr; data = data->next()) {
     if (data->is_alive()) {
```

Comment:
There used to be more code for the case where data->is_alive() so this code had this continue, but you could simplify this by having an 'else' clause and remove the continue. That would clean this up a little.

Reply:
Fixed